core/src/main/java/org/elasticsearch/common/geo/ShapeRelation.java

```diff
@@ -65,4 +65,8 @@ public static ShapeRelation getRelationByName(String name) {
     public String getRelationName() {
         return relationName;
     }
+
+    public boolean isAllowedInRangeQuery() {
+        return this == INTERSECTS || this == CONTAINS || this == WITHIN;
+    }
```

Comment:
It feels a bit wrong to me to make ShapeRelation aware of how it is used in range queries. I'd rather put this logic in RangeQueryBuilder.

Reply:
Make sense! Very thanks!